pandas/tests/io/pytables/test_store.py

```diff
@@ -4888,6 +4888,50 @@ def test_unsuppored_hdf_file_error(self, datapath):
         with pytest.raises(ValueError, match=message):
             pd.read_hdf(data_path)
 
+    def test_supported_for_subclasses_dataframe(self):
```

Comment:
put in a new file test_subclass.py

Reply:
Moved to a separate file.

Right, ``read_hdf`` does not recreate the subclass. It seems that it would return the ``DataFrame`` with the same content.